transports/webrtc/src/in_addr.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use if_watch::{IfEvent, IfWatcher};
+
+use futures::{
+    future::{BoxFuture, FutureExt},
+    stream::Stream,
+};
+
+use std::{
+    io::Result,
+    net::IpAddr,
+    ops::DerefMut,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// Watches for interface changes.
+#[derive(Debug)]
+pub enum InAddr {
```

Comment:
From a first quick look it seems like this is copied from the tcp (async_io) `InAddr`. Since the same is needed for quic, we are considering to extract it into a separate place from where all three transports (Tcp, Quic and WebRTC) can import it from. See this discussion: https://github.com/libp2p/rust-libp2p/pull/2289#discussion_r917403346.
Would that work for you or does your logic differ from the tcp one here?

Reply:
> Would that work for you or does your logic differ from the tcp one here?

It's the same. Extracting it sounds great 👍

Reply:
With `IfWatcher` version `2.0.0` that was published recently, I think extracting the logic is not necessary anymore as the integration became much simpler. See corresponding PR that updates the TCP transport #2813.